oeps/best-practices/oep-0011/decisions/0007-bundle-size.rst

```diff
@@ -0,0 +1,55 @@
+Use BundleWatch to automate bundle size checking
+################################################
+
+Status
+******
+
+Accepted
+
+Context
+*******
+
+It is important for users of the Open edX platform that we deliver reasonably sized JavaScript bundles. This provides
+faster load times to all users, and is vital for users with low bandwidth and/or metered connections. To ensure we 
+don't unintentionally increase the size of our JavaScript bundles, we should utilize automated bundle size monitoring.
```

Comment:
[curious] Is it worth mentioning that we have _some_ support for bundle size analytics today insofar that we do generate a bundle size report via the standard Webpack configurations provided by frontend-build which can be viewed locally? E.g., when you run `npm run build`, a `report.html` file is generated in the `dist` directory that lets developers see bundle sizes of JS assets (compressed vs. uncompressed). However, this report is not very discoverable. Exposing similar data via these CI checks makes bundle size a more prominent concern than with the tools Open edX engineers have to work with today.

Reply:
I agree it's a good idea to mention it: I myself had totally forgotten about that report!

Reply:
added in https://github.com/openedx/open-edx-proposals/compare/73eae79af86499f83d01a126913ee59caed1f72d..9ffa820134819769b9b937af06c4bc6e39c4235c